File: StackOverflow/stack_overflow.c
#include <stdio.h>

int fonction(int n){
	int y;
	if(n<=0){
		return 0;
	}
	
	if(n>=1){
		return 1+fonction(n-1);
	}
}

int main(void){
	int x = 1000;
	int y = 1000000;
	printf("%d",fonction(x));
	printf("%d",fonction(y));
}

File: StackOverflow/stack_overflow.java
public class Fonction{
	
	public Fonction(){
	}
	
	public int f(int n){
		if(n<=0){
			return 0;
		}
		
		if(n>=1){
			return 1+f(n-1);
		} 
	}	
}

public static void main(){
	Fonction fct= new Fonction();
	fct.f(1000);
}
